sgv2-service-common/src/main/java/io/stargate/sgv2/common/cql/builder/Column.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright DataStax, Inc. and/or The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.sgv2.common.cql.builder;
+
+import io.stargate.sgv2.common.cql.ColumnUtils;
+import javax.annotation.Nullable;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Immutable;
+
+@Immutable(prehash = true)
+public interface Column {
+  String name();
+
+  /** TODO do we ever need a more structured representation? */
```

Comment:
what would `a more structured representation` be?

Reply:
Something like `ColumnType` in the persistence API.
Currently we are focused on the REST API, where types are just provided as strings in the JSON payload, so this is good enough.
But for GraphQL we have more elaborate use cases, like translating a GraphQL type into a CQL type.

Reply:
Got it.. Yea maybe we change when we need it, or?

Reply:
I wonder if we need "type builder" (and perhaps -- but not necessarily -- "type decoder")?
What I mean is that when building type signatures like `frozen<map<uuid, tinyint>>` one could have simple builder, used optionally for structured cases.

I agree that this could be added at a later stage for sure; but I also suspect that SGv2 REST API would quite soon need it when converting between gRPC protoc and REST JSON.